lib/ansible/modules/windows/win_blockinfile.ps1

```diff
@@ -0,0 +1,330 @@
+#!powershell
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+# Large portions borrowed from Brian Lloyd's (@brianlloyd) win_lineinfile 
+# and Yaegashi Takeshi's (@yaegashi) blockinfile
+
+function WriteLines($outlines, $path, $linesep, $encodingobj, $validate, $check_mode) {
+
+    $acl = $null
+
+    Try {
+        If (Test-Path -LiteralPath $path) {
+            $acl = Get-Acl -Path $path
+        }
+    }
+    Catch {
+        Fail-Json @{} "Cannot get ACL from original file temporary file! ($($_.Exception.Message))";
+    }
+    Try {
+        $temppath = [System.IO.Path]::GetTempFileName();
+    }
+    Catch {
+        Fail-Json @{} "Cannot create temporary file! ($($_.Exception.Message))";
+    }
+    $joined = $outlines -join $linesep;
+    [System.IO.File]::WriteAllText($temppath, $joined, $encodingobj);
+
+    If ($validate) {
+
+        If (-not ($validate -like "*%s*")) {
+            Fail-Json @{} "validate must contain %s: $validate";
+        }
+
+        $validate = $validate.Replace("%s", $temppath);
+
+        $parts = [System.Collections.ArrayList] $validate.Split(" ");
+        $cmdname = $parts[0];
+
+        $cmdargs = $validate.Substring($cmdname.Length + 1);
+
+        $process = [Diagnostics.Process]::Start($cmdname, $cmdargs);
+        $process.WaitForExit();
+
+        If ($process.ExitCode -ne 0) {
+            [string] $output = $process.StandardOutput.ReadToEnd();
+            [string] $error = $process.StandardError.ReadToEnd();
+            Remove-Item $temppath -force;
+            Fail-Json @{} "failed to validate $cmdname $cmdargs with error: $output $error";
+        }
+
+    }
+
+    # Commit changes to the path
+    $cleanpath = $path.Replace("/", "\");
+    Try {
+        Copy-Item $temppath $cleanpath -force -ErrorAction Stop -WhatIf:$check_mode;
+    }
+    Catch {
+        Fail-Json @{} "Cannot write to: $cleanpath ($($_.Exception.Message))";
+    }
+
+    If ($acl) {
+        Try {
+            Set-Acl -Path $cleanpath -AclObject $acl -WhatIf:$check_mode 
+        }
+        Catch {
+            Fail-Json @{} "Cannot set ACL on new file! ($($_.Exception.Message))";
+        }
+    }
+
+    Try {
+        Remove-Item $temppath -force -ErrorAction Stop;
+    }
+    Catch {
+        Fail-Json @{} "Cannot remove temporary file: $temppath ($($_.Exception.Message))";
+    }
+
+    return $joined;
+}
+
+
+# Backup the file specified with a date/time filename
+function BackupFile($path, $check_mode) {
```

Comment:
We already have a Backup-File implementation in https://github.com/ansible/ansible/pull/50033
It would be nice if this PR would (eventually) use the new interface.

Reply:
I subscribed to that PR, when it merges I'll switch it over.

Reply:
Updated to use shared `Backup-File` implemenation.